src/core/internal/elf/dl.d

```diff
@@ -112,6 +120,42 @@ struct SharedObject
         return cstr[0 .. strlen(cstr)];
     }
 
+    /**
+     * Tries to fill the specified buffer with the path to the ELF file,
+     * according to the /proc/<PID>/maps file.
+     *
+     * Returns: The filled slice (null-terminated), or null if an error occurs.
+     */
+    char[] getPath(size_t N)(ref char[N] buffer) const
+    if (N > 1)
+    {
+        import core.stdc.stdio, core.stdc.string, core.sys.posix.unistd;
+
+        char[N + 128] lineBuffer = void;
+
+        snprintf(lineBuffer.ptr, lineBuffer.length, "/proc/%d/maps", getpid());
```

Comment:
In FreeBSD, using `kinfo_getvmmap` seems more suitable, this works even if /proc is not mounted.
https://www.freebsd.org/cgi/man.cgi?query=kinfo_getvmmap

Reply:
I'm not sure that's worth it (incl. the `libutil` dependency); I've just tried to write this in a portable way (e.g., no /proc/**self**/maps etc.).